### DISCUSSION TITLE
movement interactions

### DISCUSSION THREAD

**Discussion by xLink** [Annoraaq/grid-engine#190] [Ideas]
not entirely sure if this is in scope of this engine or not, but there are 2 diff types of movement im looking to implement:

Ice Movement, fairly common in these top down games, you move onto a tile, and carry on moving until you hit a collision, walls, rocks etc
![image](https://user-images.githubusercontent.com/217433/131250434-de7a76da-a612-4c4b-91cf-d2463ad743ea.png)

Spin tiles, move onto a tile, and follow the path until you hit a stop tile or a new movement tile.
![image](https://user-images.githubusercontent.com/217433/131250482-ed72f04d-ed7b-49c4-85e9-678b4253725f.png)
![image](https://user-images.githubusercontent.com/217433/131250700-f95c83cb-afad-4c87-bccc-3034676e3338.png)

These open the door for creating puzzles in the 2d world, and would be a great addition to the library.

**Comment by Annoraaq** [comment]
I agree. They should already be possible with GridEngine but it might be a good idea to provide them out of the box.

**Comment by xLink** [comment]
For anyone seeing this now, all 3 of these should be viable now, there is an [Ice Movement](https://annoraaq.github.io/grid-engine/examples/ice-movement.html) tutorial on the wiki that ive used to make my ice and spinner tiles, havent tried the water version yet, but i can prolly use the same idea for the spin tiles without too much issue. :D
